File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.js
function solution(queue1, queue2) {
    let q1Sum = 0, q2Sum = 0;
    
    for (let i = 0; i < queue1.length; i++) {
        q1Sum += queue1[i];
        q2Sum += queue2[i];
    }
    
    const HALF = (q1Sum + q2Sum) / 2;
    
    const queue = [...queue1, ...queue2];
    const queueLength = queue.length;
    
    let pointer1 = 0, pointer2 = queue1.length;
    
    for(let i = 0; i < queue1.length * 3; i++) {
        if(q1Sum === HALF) return i;
        if(HALF < q1Sum) q1Sum -= queue[pointer1++ % queueLength];
        else q1Sum += queue[pointer2++ % queueLength];
    }
    
    return -1;
}

File: 프로그래머스/2/118667. 두 큐 합 같게 만들기/두 큐 합 같게 만들기.kt
class Solution {
    fun solution(queue1: IntArray, queue2: IntArray): Int {
        var q1Sum:Long = queue1.sum().toLong()
        var q2Sum:Long = queue2.sum().toLong()
        
        val half = (q1Sum + q2Sum) / 2
        
        val queue = queue1 + queue2
        val queueLength = queue.size
        
        var p1 = 0
        var p2 = queue1.size
        
        for(cnt in 0 until queue1.size * 3) {
            if(q1Sum == half) {
                return cnt
            }
            
            if(half < q1Sum) {
                q1Sum -= queue[p1++ % queueLength]
            }
            else {
                q1Sum += queue[p2++ % queueLength]
            }
        }
        return -1;
    }
}